src/scripts/main.js

```diff
@@ -1 +1,13 @@
 'use strict';
+
+const ul = document.querySelector('ul');
+const target = document.getElementById('largeImg');
+
+ul.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const link = e.target.closest('.list-item__link');
+  const href = new URL(link.getAttribute('href'), window.location.origin).href;
```

Comment:
If `link` is `null`, `link.getAttribute('href')` will throw an error. Ensure that `link` is not `null` before attempting to access its attributes.